app/controllers/circle/projects_controller.rb

```diff
@@ -61,6 +61,7 @@ def update
 
   def complete
     authorize! :complete, current_project
+    close_all_tasks_supplies
```

Comment:
There are a few advantages to moving the business logic to mutations (or more general, separate objects), including easier reuse and testability.